### PR TITLE
Make mocking configuration less confusing.

### PR DESCRIPTION
**Pull request by deftomat**
Setting mocks to false will always disable mocking.

We had a few WTF? situations  😉  with the following Apollo server config:

```js
const config = {
  typeDefs,
  resolvers,
  mocks: false,
  mockEntireSchema: false,
}
```

Everyone in our team assumed that you can disable the mocks by setting the `mocks` property to false.
However, according to:
https://github.com/apollographql/apollo-server/blob/a12673a432e01e5f4922349c38bbf7fc55e6628b/packages/apollo-server-core/src/ApolloServer.ts#L257

The mocks will be enabled when you provide `mocksEntireSchema` together with `mocks`. It doesn't matter if `mocks` is `true` or `false` or an object.

I'm not sure if this is an intended behaviour but I can definitely say that it is really confusing.
Can we update it to disable mocks when `mocks` property is `false`?

(Feel free to close this PR if the current behaviour is an intended behaviour)

### CODE REVIEW

File: packages/apollo-server-core/src/ApolloServer.ts [hunk 1]
```diff
@@ -254,7 +254,7 @@ export class ApolloServerBase {
       });
     }
 
-    if (mocks || typeof mockEntireSchema !== 'undefined') {
+    if (mocks || (typeof mockEntireSchema !== 'undefined' && mocks !== false)) {
```
[the remark this answers is not in the document]
Comment (deftomat) [reply packages/apollo-server-core/src/ApolloServer.ts]: Well, in my code, you can enable mocks by setting `mockEntireSchema` without bothering about `mocks` property. 
By looking at the next lines, you can see that it is an intended behaviour to be able to enable mocks just by using `mockEntireSchema`.

**Comment by JakeDawkins** [comment]
@deftomat Can you add this line to `CHANGELOG.md` for me? FYI, you may need to pull first also, since I updated the branch

```
- Fix mocks configuration to allow disabling of mocks by using `mocks: false`, even if `mockEntireSchema` is `true` [PR #1835](https://github.com/apollographql/apollo-server/pull/1835)
```

**Comment by deftomat** [comment]
@JakeDawkins done!

**Comment by JakeDawkins** [comment]
Thanks for the help @deftomat!

**Pull request merged**